h5netcdf/tests/test_h5netcdf.py

```diff
@@ -101,223 +111,234 @@ def is_h5py_char_working(tmp_netcdf, name):
+    assert v.dimensions == ("x", "y")
     assert v.ndim == 2
-    assert v.ncattrs() == ['units']
-    if not PY2 and write_module is not netCDF4:
-        assert v.getncattr('units') == 'meters'
+    assert v.ncattrs() == ["units"]
+    if write_module is not netCDF4:
+        assert v.getncattr("units") == "meters"
     assert tuple(v.chunking()) == (4, 5)
-    assert v.filters() == {'complevel': 4, 'fletcher32': False,
-                           'shuffle': True, 'zlib': True}
-
-    v = ds.variables['y']
+    assert v.filters() == {
+        "complevel": 4,
+        "fletcher32": False,
+        "shuffle": True,
+        "zlib": True,
+    }
+
+    v = ds.variables["y"]
     assert array_equal(v, np.r_[np.arange(4), [-1]])
     assert v.dtype == int
-    assert v.dimensions == ('y',)
+    assert v.dimensions == ("y",)
     assert v.ndim == 1
-    assert v.ncattrs() == ['_FillValue']
-    assert v.getncattr('_FillValue') == -1
-    assert v.chunking() == 'contiguous'
-    assert v.filters() == {'complevel': 0, 'fletcher32': False,
-                           'shuffle': False, 'zlib': False}
+    assert v.ncattrs() == ["_FillValue"]
+    assert v.getncattr("_FillValue") == -1
+    assert v.chunking() == "contiguous"
+    assert v.filters() == {
+        "complevel": 0,
+        "fletcher32": False,
+        "shuffle": False,
+        "zlib": False,
+    }
     ds.close()
 
     # Check the behavior if h5py. Cannot expect h5netcdf to overcome these
     # errors:
-    if is_h5py_char_working(tmp_netcdf, 'z'):
-        ds = read_module.Dataset(tmp_netcdf, 'r')
-        v = ds.variables['z']
+    if is_h5py_char_working(tmp_netcdf, "z"):
+        ds = read_module.Dataset(tmp_netcdf, "r")
+        v = ds.variables["z"]
         assert array_equal(v, _char_array)
-        assert v.dtype == 'S1'
+        assert v.dtype == "S1"
         assert v.ndim == 2
-        assert v.dimensions == ('z', 'string3')
-        assert v.ncattrs() == ['_FillValue']
-        assert v.getncattr('_FillValue') == b'X'
+        assert v.dimensions == ("z", "string3")
+        assert v.ncattrs() == ["_FillValue"]
+        assert v.getncattr("_FillValue") == b"X"
     else:
-        ds = read_module.Dataset(tmp_netcdf, 'r')
+        ds = read_module.Dataset(tmp_netcdf, "r")
 
-    v = ds.variables['scalar']
+    v = ds.variables["scalar"]
     assert array_equal(v, np.array(2.0))
-    assert v.dtype == 'float32'
+    assert v.dtype == "float32"
     assert v.ndim == 0
     assert v.dimensions == ()
     assert v.ncattrs() == []
 
-    v = ds.variables['intscalar']
+    v = ds.variables["intscalar"]
     assert array_equal(v, np.array(2))
-    assert v.dtype == 'int64'
+    assert v.dtype == "int64"
     assert v.ndim == 0
     assert v.dimensions == ()
     assert v.ncattrs() == []
 
-    v = ds.variables['var_len_str']
-    assert v.dtype == str
-    assert v[0] == u'foo'
+    v = ds.variables["var_len_str"]
+    print("dtype:", v.dtype)
+    # assert v.dtype == 'O'
```

Comment:
This is the test for the commented out inline setting "str" as a dtype

Reply:
And forgot this one too...